pkg/cloud/ovirt/clients/machineservice.go

```diff
@@ -153,12 +157,20 @@ func (is *InstanceService) InstanceCreate(
 	_, err = is.Connection.SystemService().VmsService().
 		VmService(response.MustVm().MustId()).
 		TagsService().Add().
-		Tag(ovirtsdk.NewTagBuilder().Name(machine.Labels["machine.openshift.io/cluster-api-cluster"]).MustBuild()).
+		Tag(ovirtsdk.NewTagBuilder().Name(ovirtClusterID).MustBuild()).
 		Send()
 	if err != nil {
 		klog.Errorf("Failed to add tag to VM, skipping", err)
 	}
 
+	err = is.handleAffinityGroup(
+		response.MustVm(),
+		providerSpec.ClusterId,
+		providerSpec.AffinityGroupsNames)
+	if err != nil {
+		klog.Errorf("Failed to add VM to AffinityGroup, skipping", err)
```

Comment:
This currently treats the affinity groups as "optional". If it fails, it still goes through with VM creation. This is bad because nobody I know operating an OpenShift cluster will look at those logs except something goes wrong and affinity groups are a hard requirement for things like redundant databases.

Reply:
Done